src/lib/server/hyrise_session.cpp

```diff
@@ -23,278 +23,270 @@
-  CurrentScheduler::schedule_tasks(tasks);
+//  std::vector<std::shared_ptr<AbstractTask>> tasks = {std::make_shared<ExecuteServerQueryTask>(*_sql_pipeline)};
+//  CurrentScheduler::schedule_tasks(tasks);
 }
 
 void HyriseSession::query_executed() {
-  const std::vector<std::shared_ptr<SendQueryResponseTask>> tasks = {
-      std::make_shared<SendQueryResponseTask>(_self, *_sql_pipeline, _sql_pipeline->get_result_table())};
-  CurrentScheduler::schedule_tasks(tasks);
+//  const std::vector<std::shared_ptr<SendQueryResponseTask>> tasks = {
+//      std::make_shared<SendQueryResponseTask>(_connection, *_sql_pipeline, _sql_pipeline->get_result_table())};
+//  CurrentScheduler::schedule_tasks(tasks);
 }
 
 void HyriseSession::prepared_bound(std::unique_ptr<SQLQueryPlan> query_plan,
                                    std::shared_ptr<TransactionContext> transaction_context) {
   _prepared_query_plan = std::move(query_plan);
   _prepared_transaction_context = std::move(transaction_context);
   auto output_packet = PostgresWireHandler::new_output_packet(NetworkMessageType::BindComplete);
-  async_send_packet(output_packet);
-  return _async_receive_header();
+//  async_send_packet(*output_packet);
+//  return _async_receive_header();
 }
 
 void HyriseSession::prepared_executed(std::shared_ptr<const Table> result_table) {
-  const std::vector<std::shared_ptr<SendQueryResponseTask>> tasks = {
-      std::make_shared<SendQueryResponseTask>(_self, *_sql_pipeline, std::move(result_table))};
-  CurrentScheduler::schedule_tasks(tasks);
+//  const std::vector<std::shared_ptr<SendQueryResponseTask>> tasks = {
+//      std::make_shared<SendQueryResponseTask>(_connection, *_sql_pipeline, std::move(result_table))};
+//  CurrentScheduler::schedule_tasks(tasks);
 }
 
-void HyriseSession::query_response_sent() { _send_ready_for_query(); }
+void HyriseSession::query_response_sent() { 
+//  _send_ready_for_query(); 
+}
 
 void HyriseSession::load_table_file(const std::string& file_name, const std::string& table_name) {
-  const std::vector<std::shared_ptr<LoadServerFileTask>> tasks = {
-      std::make_shared<LoadServerFileTask>(_self, file_name, table_name)};
-  CurrentScheduler::schedule_tasks(tasks);
+//  const std::vector<std::shared_ptr<LoadServerFileTask>> tasks = {
+//      std::make_shared<LoadServerFileTask>(file_name, table_name)};
+//  CurrentScheduler::schedule_tasks(tasks);
 }
 
 void HyriseSession::pipeline_error(const std::string& error_msg) {
-  _send_error(error_msg);
-  _send_ready_for_query();
+//  _send_error(error_msg);
+//  _send_ready_for_query();
 }
 
-SessionState HyriseSession::state() const { return _state; }
+//SessionState HyriseSession::state() const { return _state; }
+
+template<typename T>
+auto HyriseSession::_dispatch_server_task(std::shared_ptr<T> task) -> decltype(task->get_future()) {
+  using TaskList = std::vector<std::shared_ptr<AbstractTask>>;
+
+  CurrentScheduler::schedule_tasks(TaskList({task}));
```

Comment:
i think we can omit  `TaskList({task})` and only write `{task}`

Reply:
I've tried -- we can't :(